localstack-core/localstack/services/lambda_/invocation/execution_environment.py

```diff
@@ -343,11 +341,22 @@ def get_prefixed_logs(self) -> str:
 
     def invoke(self, invocation: Invocation) -> InvocationResult:
         assert self.status == RuntimeStatus.RUNNING
+        # Async/event invokes might miss an aws_trace_header, then we need to create a new root trace id.
+        aws_trace_header = (
+            invocation.trace_context.get("aws_trace_header") or TraceHeader().ensure_root_exists()
+        )
+        # The Lambda RIE requires a full tracing header including Root, Parent, and Samples. Otherwise, tracing fails
+        # with the warning "Subsegment ## handler discarded due to Lambda worker still initializing"
+        aws_trace_header.ensure_sampled_exists()
+        # TODO: replace this random parent id with actual parent segment created within the Lambda provider using X-Ray
+        aws_trace_header.ensure_parent_exists()
+        # TODO: test and implement Active and PassThrough tracing and sampling decisions.
+        # TODO: implement Lambda lineage: https://docs.aws.amazon.com/lambda/latest/dg/invocation-recursion.html
```

Comment:
Thanks for leaving the TODOs here. Those seem like really interesting tasks for engineers getting started at LocalStack :wink: